src/components/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,73 @@
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { faCaretDown, faTools } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { pages } from '../../assets/pages';
+
+import './NavBar.css';
+import NavBarTools from './NavBarTools';
+import NavBarLogo from './NavBarLogo';
+
+interface PlatformType {
+  id: string;
+  header: string;
+  style: string;
+}
+interface NavBarPropsType {
+  toggleMainWidth: () => void;
+  mini: Boolean;
+  platform: PlatformType;
+}
+
+const NavBar = ({ toggleMainWidth, mini, platform }: NavBarPropsType) => {
+  const [sideBarWidthClass, setSideBarWidthClass] = useState('sidebar-min');
+
+  const toggleSideBarWidth = (isMouseEvent: Boolean) => {
+    if (isMouseEvent) {
+      toggleMainWidth();
+      setSideBarWidthClass(mini ? 'sidebar-max' : 'sidebar-min');
+    }
+  };
+
+  return (
+    <div>
+      <nav
+        className={`sidebar ${platform.style} ${sideBarWidthClass}`}
+        role="navigation"
+        onMouseOver={() => toggleSideBarWidth(true)}
+        onFocus={() => toggleSideBarWidth(false)}
+        onMouseOut={() => toggleSideBarWidth(true)}
+        onBlur={() => toggleSideBarWidth(false)}
+      >
+        <NavBarLogo header={platform.header} />
+        <div className="navbar-items">
+          {pages
```

Comment:
not sure if to this PR but

why not impmant routings that you'll have src(home) than platform than specific page?
like 'https://HUB/F15/Py-Build'

and than the separation will be in the routing level and not here?

Reply:
The tools can belong to several platforms and I dont want to seperate them this way.
The platforms are only meant for filtering the navbar options (display only).
In other words, each tool should be logically accessible regardless of the user's chosen platform